core/include/detray/core/detector.hpp

```diff
@@ -167,39 +171,54 @@ class detector {
         return cvolume;
     }
 
+    template <typename detector_data_type,
+              std::enable_if_t<!std::is_base_of_v<vecmem::memory_resource,
+                                                  detector_data_type>,
+                               bool> = true>
+    DETRAY_DEVICE detector(detector_data_type &det_data)
```

Comment:
Don't make it only device code. We should allow creating a `detray::detector` object on top of a "data object" in host code as well. We should actually add a couple of tests for that later on as well. Testing that these objects can be constructed on top of a "data object" should not be the exclusive domain of the device/CUDA tests. :wink:

Reply:
> We should allow creating a detray::detector object on top of a "data object" in host code as well. 

Hmm. What could be the use case of this?

Reply:
I made it `detray_host_device` for the moment